app/lib/katello/validators/environment_docker_repositories_validator.rb

```diff
@@ -0,0 +1,78 @@
+module Katello
+  module Validators
+    class EnvironmentDockerRepositoriesValidator < ActiveModel::Validator
+      def validate(environment)
+        return true if environment.registry_name_pattern.empty?
+
+        unless ContainerImageNameValidator.validate_name(Katello::Repository.safe_render_container_name(test_repository, environment.registry_name_pattern))
+          environment.errors.add(:registry_name_pattern, N_("Registry name pattern will result in invalid container image name of member repositories"))
+          return false
+        end
+
+        error_messages = EnvironmentDockerRepositoriesValidator.validate_repositories(environment, environment.registry_name_pattern, environment.repositories.where(content_type: Katello::Repository::DOCKER_TYPE))
```

Comment:
can change: 
`environment.repositories.where(content_type: Katello::Repository::DOCKER_TYPE) `
to
`environment.repositories.docker_type`

Reply:
and actually, can't you just pass in environment, and let the method call environment.repositories.docker_type (kind of a nitpick)

Reply:
Passing in the repositories directly is used in ContentView model. I will switch to .docker_type